didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+            packed = await resp.text()
+
+            # If the HTTP enpoint responded with a message, decode it
+            if len(packed) > 0:
+                unpacked = await dmp.packaging.unpack(packed)
+                msg = unpacked[0].decode()
+                LOG.debug("Raw message from remote %s" % msg)
+                return Message.from_json(msg)
+    return
+
+
+async def setup_relay(
+    dmp: DIDCommMessaging, my_did: DID, relay_did: DID, keys: Sequence[Key]
+) -> Union[DID, None]:
+    """Negotiate services with an outbound relay.
+
+    Returns a DID upon successful negotiation.
+    """
+
+    # Request mediation from the outbound relay
+    message = Message(
+        type="https://didcomm.org/coordinate-mediation/3.0/mediate-request",
+        id=str(uuid.uuid4()),
+        body={},
+        frm=my_did,
+        to=[relay_did],
+    )
+    message = await send_http_message(dmp, my_did, message, target=relay_did)
+
+    # Verify that mediation-access has been granted
+    if message.type == "https://didcomm.org/coordinate-mediation/3.0/mediate-deny":
+        return
+    if message.type != "https://didcomm.org/coordinate-mediation/3.0/mediate-grant":
+        # We shouldn't run into this case, but it's possible
+        raise Exception("Unknown response type received: %s" % message.type)
+
+    # Create a new DID with an updated service endpoint, pointing to our relay
+    relay_did = message.body["routing_did"][0]
+    new_did = generate(
+        keys,
+        [
+            {
+                "type": "DIDCommMessaging",
+                "serviceEndpoint": {
+                    "uri": relay_did,
+                    "accept": ["didcomm/v2"],
+                },
+            }
+        ],
+    )
+    LOG.info("relayed did: ", new_did)
+
+    # A couple of helpers variables to simplify the next few lines
+    resolver = dmp.resolver
+    secrets = dmp.secrets
+
+    # Add the DID to our secrets manager so that we can unpack messages
+    # destined to us via our new DID
+    doc = await resolver.resolve_and_parse(new_did)
+    # New format, key-# is in order of keys in did
+    await secrets.add_secret(AskarSecretKey(keys[0], f"{new_did}#key-1"))
+    await secrets.add_secret(AskarSecretKey(keys[1], f"{new_did}#key-2"))
+
+    # Legacy formats
+    # verkey
+    await secrets.add_secret(AskarSecretKey(keys[0], doc.authentication[0]))
+    # xkey
+    await secrets.add_secret(AskarSecretKey(keys[1], doc.key_agreement[0]))
+
+    # Send a message to the relay informing it of our new endpoint that people
+    # should contact us by
+    message = Message(
+        type="https://didcomm.org/coordinate-mediation/3.0/recipient-update",
+        id=str(uuid.uuid4()),
+        body={
+            "updates": [
+                {
+                    "recipient_did": new_did,
+                    "action": "add",
+                },
+            ],
+        },
+        frm=my_did,
+        to=[relay_did],
+    )
```

Comment:
I used to have a strong preference towards strongly typed message structures all the time. I find my preferences more nuanced now. For the purposes of a quick start, I recommend just using plain old dictionaries instead of a Message class. I think using dictionaries will make it easier to map between examples in DIDComm protocol definitions and code as well as avoiding the confusion of python keywords like `from` interfering with the attribute name. I think this also comes with the benefit of just being simpler. I believe in BYOM (bring your own models) lol.